app/models/subnet/ipv4.rb

```diff
@@ -35,6 +35,32 @@ def cleanup_addresses
     self
   end
 
+  # imports subnets from a dhcp smart proxy
+  def self.import(proxy)
+    return unless proxy.has_feature?('DHCP')
+    ProxyAPI::DHCP.new(:url => proxy.url).subnets.map do |s|
+      # do not import existing networks.
+      attrs = { :network => s["network"], :mask => s["netmask"] }
+
+      next if exists?(attrs)
+      attrs.merge!(parse_dhcp_options(s['options'])) if s['options'].present?
+      new(attrs.update(:dhcp => proxy))
+    end.compact
+  end
+
+  # Translate ISC dhcp subnet options names provided by dhcp proxy into foreman subnet attributes names
+  def self.parse_dhcp_options(options)
+    attrs = {}
+    attrs[:gateway]          = options["routers"][0]             if options["routers"] && options["routers"][0]
+    attrs[:dns_primary]      = options["domain_name_servers"][0] if options["domain_name_servers"] && options["domain_name_servers"][0]
+    attrs[:dns_secondary]    = options["domain_name_servers"][1] if options["domain_name_servers"] && options["domain_name_servers"][1]
+    attrs[:from], attrs[:to] = options["range"]                  if options["range"] && options["range"][0] && options["range"][1]
+
+    attrs
+  end
+
+  private_class_method :parse_dhcp_options
```

Comment:
Ah, then defining `class << self` might be a better option then. It's always confusing to have class methods & properties scattered